astropy/io/registry/tests/test_registries.py

```diff
@@ -353,6 +353,7 @@ def test_delay_doc_updates(self, registry, fmtcls1):
                     assert docs[-1][iread : iread + 3] != "Yes"
         # now test it's updated
         docs = EmptyData.read.__doc__.split("\n")
+        ihd = [i for i, s in enumerate(docs) if ("Format" in s)][0]
```

Comment:
I think a slightly larger factor is called for here. The above test block could be dedented from the context and then some of the index creators can be moved out of the conditional block using the solution here.

Reply:
> The above test block could be dedented from the context

I'm not sure that's the case; without looking to deeply into how this context works exactly, it seems to me that the only difference between line 346 and line 355 is that the former happens within the context, while the latter doesn't.

Reply:
Hm. you're right. It happens at context exit.